docs/release/release-tasks.md

```diff
@@ -310,30 +310,38 @@ The goal of this task to make the book for the current release available under e
    <br>Prior art: [Add release 1.2 book link](https://github.com/kubernetes-sigs/cluster-api/pull/6697)
 
 #### Create PR for release notes
-
+1. Checkout the `main` branch.
+1. Build the release note tools binary.
+   ```bash
+   go build -o bin/notes hack/tools/release/notes.go
+   ```
```

Comment:
❓ : `make release-notes` has special handling about e.g. `PRE_RELEASE` to add some text as header:

https://github.com/kubernetes-sigs/cluster-api/blob/1735aa5550db514a0dd00b19a1d307e07766b885/Makefile#L1086-L1091

 I may not be up to date on the tooling but shouldn't we be able to do sth like the following (and by that be able to skip the step of compiling the notes binary?)
  
 ```sh
 RELEASE_NOTES_DIR=CHANGELOG
 PREVIOUS_TAG=v1.3.x
 PRE_RELEASE=y
 RELEASE_TAG=v1.4.x
 make release-notes
 ```
 
 This way we would make sure (after checking out main) that we always use the latest version of the tool and not be able to be lazy and skip building the new version.

Reply:
The problem with this is the tool needs to be run on the branch you are releasing from. If we want to do this, we need to backport all release notes tool improvements to the release branches so they are available when running the target. Compiling to binary from main allows to avoid this process.

We have an improvement task to be able to run the tool from any branch by reading directly  from github instead of the local git. But we haven't got to it yet.